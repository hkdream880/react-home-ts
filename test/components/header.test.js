import React from 'react'
import ReactDOM from 'react-dom'
import Header from '@/components/header'


import { render, cleanup } from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"

import renderer from 'react-test-renderer'

// it('render without crashing', () => {
//   const div = document.createElement('div')
//   ReactDOM.render(<Header />, div)
// })

afterEach(cleanup)
it('헤더가 있는가?', () => {
  const { getByTestId } = render(<Header />)
  expect(getByTestId('header')).toBeDefined()
})

it('스냅샷 테스트', () => {
  const snap_tree = renderer.create(<Header />).toJSON()
  expect(snap_tree).toMatchSnapshot();
})