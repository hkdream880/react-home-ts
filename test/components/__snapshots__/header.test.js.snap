// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`헤더 테스트 입니다. 스냅샷 테스트 1`] = `
<div
  className="makeStyles-root-4"
  data-testid="header"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-5 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
      <h6
        className="MuiTypography-root makeStyles-title-6 MuiTypography-h6"
      >
        News
      </h6>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Login
        </span>
      </button>
    </div>
  </header>
</div>
`;
